src/mashupDocumentParser.ts

```diff
@@ -56,4 +66,83 @@ export default class MashupHandler {
 
         return section1m;
     };
+
+    
+    private editSingleQueryMetadata = (metadataArray: Uint8Array, metadata: Metadata) => {
```

Comment:
This area needs testing, try and extract the logic you have for entries and item paths to a method and unit test that.

Reply:
Saw the test in next iteration (👍), gave some comments there.